repos/system_upgrade/el7toel8/actors/sssdcheck/actor.py

```diff
@@ -0,0 +1,98 @@
+
+
+class SSSDCheck(Actor):
+    '''
+    Check SSSD configuration for changes in RHEL8 and report them.
+    
+    These changes are:
+    - id_provider=local is no longer supported and will be ignored
+    - ldap_groups_use_matching_rule_in_chain was removed and will be ignored
+    - ldap_initgroups_use_matching_rule_in_chain was removed and will be ignored
+    - ldap_sudo_include_regexp changed default from true to false
+    '''
+    
+    name = 'sssd_check'
+    consumes = ()
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+    
+    def process(self):
+        try:
+            self.config = ConfigParser.RawConfigParser()
+            self.config.read('/etc/sssd/sssd.conf')
+        except ConfigParser.Error:
+            # SSSD is not configured properly. Nothing to do.
+            return
+        
+        affected_domains = []
+        
+        for section in self.config.sections():
+            # We are interested only in domains.
+            if not section.startswith('domain/'):
+                continue
+            
+            domain = section[len('domain/'):]
+            self.checkLocalProvider(domain)
+            self.checkRemovedOption(domain, 'ldap_groups_use_matching_rule_in_chain')
+            self.checkRemovedOption(domain, 'ldap_initgroups_use_matching_rule_in_chain')
+            self.checkSudoRegexp(domain)
+
+    def checkLocalProvider(self, domain):
+        provider = self.get_provider(domain, 'id_provider')
+        if provider != 'local':
+            return
+        
+        report_generic(
+            title='SSSD Domain "%s": local provider is no longer '
+                  'supported and the domain will be ignored.' % domain,
+            summary='Local provider is no longer supported.'
+        )
+    
+    def checkRemovedOption(self, domain, option):
+        section = self.get_domain_section(domain)
+        if not self.config.has_option(section, option):
+            return
+        
+        report_generic(
+            title='SSSD Domain "%s": option %s has no longer ' 
+                  'any effect' % (domain, option),
+            summary='Option %s was removed and it will be ignored.' % option
+        ) 
+
+    def checkSudoRegexp(self, domain):
+        section = self.get_domain_section(domain)
+        provider = self.get_provider(domain, 'sudo_provider', ['id_provider'])
+        
+        if provider not in ['ad', 'ldap']:
+            return
+
+        if not self.config.has_option(section, 'ldap_sudo_include_regexp'):
+            return
+        
+        report_generic(
+            title='SSSD Domain "%s": sudo rules containing wildcards ' 
+                  'will stop working.' % domain,
+            summary='Default value of ldap_sudo_include_regexp changed '
+                    'from true to false for performance reason. If you use '
+                    'sudo rules with wildcards, set this option to true '
+                    'explicitly.'
```

Comment:
I am thinking whether would be better to use `report_with_remediation` function instead of this. Where the content would be modified (or split) between summary and remedaiation remediation like: remediation="If you use sudo...true explicitly".

Do you plan or have you thought about possibility to fix it automatically in future? Would you detect such rules? In case it is not possible (or it is hard) to detect such rules, maybe you would just think later about use of user dialogs, to ask user what should happened and by default set the option to true (not implemented yet, in case you are curious, make sense to create comment line like `# TODO: ANSWER_FILE` . Actually, I am curious why we do not set the value to true now explicitly? How big issue it can be for user in case they use sudo rules with wildcards? Would be the upgrade fatal for them in such cases? If so, I suggest to create even an actor with the `ApplicationsPhaseTag`  tag that will set the value to `true` explicitly and document that planned change.

Reply:
The original plan was to ask the administrator and act upon his/her choice. But unfortunately, I found a bug in SSSDConfig python API (which manipulates SSSD configuration) and this bug prevented me from using it to write the sudo option. I can not tell when this bug will be solved.

The one library I found for manipulating ini files removes comments from the config which is unacceptable for SSSD. Because ldap_sudo_include_regexp is rarely if ever used corner case, we decided that just reporting it is sufficient so I did not work on any custom parser.